src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -494,6 +498,16 @@ public new CSharpCompilationOptions WithStrongNameProvider(StrongNameProvider pr
             return new CSharpCompilationOptions(this) { StrongNameProvider = provider };
         }
 
+        public new CSharpCompilationOptions WithSyntaxTreeOptionsProvider(SyntaxTreeOptionsProvider provider)
+        {
+            if (ReferenceEquals(provider, this.SyntaxTreeOptionsProvider))
```

Comment:
@jaredpar That is not the pattern being used in the other With* methods in this class. Do you want me to update those as well?